base/path.jl

```diff
@@ -308,6 +318,26 @@ end
 
 end # os-test
 
+"""
+    joinpath(parts) -> String
+
+Join a non-empty collection of path components into a full path. If some argument is an
+absolute path or (on Windows) has a drive specification that doesn't match the drive computed for
+the join of the preceding paths, then prior components are dropped.
+
+Note on Windows since there is a current directory for each drive, `joinpath("c:", "foo")`
+represents a path relative to the current directory on drive "c:" so this is equal to "c:foo",
+not "c:\\foo". Furthermore, `joinpath` treats this as a non-absolute path and ignores the drive
+letter casing, hence `joinpath("C:\\A","c:b") = "C:\\A\\b"`.
+
+# Examples
+```jldoctest
+julia> joinpath(["/home/myuser", "example.jl"])
+"/home/myuser/example.jl"
+```
+"""
+joinpath
+
 """
     joinpath(parts::AbstractString...) -> String
```

Comment:
nit: we tend to try to DRY documentation also
```suggestion
"""
    joinpath(parts::AbstractString...) -> String
    joinpath(parts::Vector{AbstractString}) -> String
    joinpath(parts::Tuple{AbstractString}) -> String
```